### PR TITLE
Create dependabot configuration

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Related Issue

Closes #1418

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (DanielNoord) [inline .github/dependabot.yml:9]:
```suggestion
      interval: "daily"
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    labels:
+      - "dependency"
+    open-pull-requests-limit: 10
+    rebase-strategy: "disabled"
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (DanielNoord) [inline .github/dependabot.yml:18]:
```suggestion
      interval: "daily"
```

**Review by DanielNoord — CHANGES_REQUESTED** [review]
🎉

**Comment by Pierre-Sassoulas** [comment]
Outch everything updated at once kinda hurt github actions 😄 I think we'll have to relaunch one by one and take it slow

**Comment by DanielNoord** [comment]
We're just failing on concurrency for `readthedocs`. I would be okay with merging if all else passes and finally making sure that `main` builds the docs (perhaps with a manual relaunch).

**Comment by Pierre-Sassoulas** [comment]
Let's do that, except for the sphinx update

**Comment by Pierre-Sassoulas** [comment]
The coverage is also blocking, let's merge without the coverage, except for the upgrade of coverall, coverage, and pytest-cov.

**Comment by DanielNoord** [comment]
I've been cancelling some workflows on `main` (to avoid duplication). I'll let some pass when we near the end.

**Comment by Pierre-Sassoulas** [comment]
Nice I did not think of that. Maybe we could cancel everything and then rebase the last one instead ?

**Comment by DanielNoord** [comment]
Rest of the PRs can't be merged.

Btw, I don't see a `readthedocs` action on `main`. Are we sure it builds correctly from `main`?

**Comment by DanielNoord** [comment]
Never mind, apparently there is no `action` on `main`. Only on PRs. I double-checked in other repo's.

Guess we're done here 😄

**Pull request merged**